setup.py

```diff
@@ -1,9 +1,40 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright 2013-2022 The Foundry Visionmongers Ltd
-from setuptools import setup, find_packages
+from distutils.command.build_py import build_py
+import logging
+import os
+from shutil import copyfile
+from setuptools import setup
+
+import openassetio_traitgen
+
+# The python sources don't exist in this repo, but are generated via
+# openassetio-traitgen at point of build.
+# We generate them directly into the build directory structured as a
+# package.
+class GenerateThenBuild(build_py):
+    def run(self):
+        # Generate traits package directly into the package directory.
+        openassetio_traitgen.generate(
+            "traits.yml",
+            self.build_lib,
+            "python",
+            lambda _: _,
+            logging.Logger("openassetio-mediacreation-traitgen"),
+            False,
+            None,
+        )
+
+        # Move the source trait yaml to the package directory.
+        copyfile(
+            "traits.yml", os.path.join(self.build_lib, "openassetio_mediacreation", "traits.yml")
```

Comment:
This should be handled as package data. See https://setuptools.pypa.io/en/latest/userguide/datafiles.html

Reply:
We logged an issue for this : https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/issues/16

We did a short initial investigation, to see if it would "just work," but unfortunately the install was running into trouble identifying package data, I suspect related to the problems we were seeing around setuptools not figuring as as a full "package" due to generating after the fact.

But we didn't dig very far, are planning to go back later and get to it properly.